include/nuttx/compiler.h

```diff
@@ -157,6 +157,7 @@
  */
 
 #  define offsetof(a, b) __builtin_offsetof(a, b)
+#  define return_address(x) __builtin_return_address(x)
```

Comment:
I'm not sure about adding this to compiler.h. Exactly this interface is compiler specific, mostly `return_address(0)` can be implemented as inline assembly for many architectures. Like reading LR register for ARM for example.

Reply:
Inline assembly also depends on compiler too, all assembly written in c file doesn't work across the different compiler.

Reply:
Ok. Let's go with this and it can be refined based on the needs.